File: chrome/browser/content_creation/reactions/internal/android/java/src/org/chromium/chrome/browser/content_creation/reactions/ReactionServiceFactory.java
// Copyright 2021 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

package org.chromium.chrome.browser.content_creation.reactions;

import org.chromium.base.annotations.NativeMethods;
import org.chromium.chrome.browser.profiles.Profile;
import org.chromium.components.content_creation.reactions.ReactionService;

/**
 * Basic factory that creates and returns a {@link ReactionService} that is
 * attached natively to the given {@link Profile}.
 */
public class ReactionServiceFactory {
    /**
     * Used to get access to the reaction service backend.
     */
    public static ReactionService getForProfile(Profile profile) {
        return ReactionServiceFactoryJni.get().getForProfile(profile);
    }

    @NativeMethods
    interface Natives {
        ReactionService getForProfile(Profile profile);
    }
}

File: chrome/browser/content_creation/reactions/internal/reaction_service_factory.cc
// Copyright 2021 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include "chrome/browser/content_creation/reactions/internal/reaction_service_factory.h"

#include "base/memory/singleton.h"
#include "chrome/browser/profiles/profile.h"
#include "components/content_creation/reactions/core/reaction_service.h"
#include "components/keyed_service/core/keyed_service.h"
#include "content/public/browser/browser_context.h"

namespace content_creation {

// static
ReactionServiceFactory* ReactionServiceFactory::GetInstance() {
  return base::Singleton<ReactionServiceFactory>::get();
}

// static
ReactionService* ReactionServiceFactory::GetForProfile(Profile* profile) {
  return static_cast<ReactionService*>(
      GetInstance()->GetServiceForBrowserContext(profile, /*create=*/true));
}

ReactionServiceFactory::ReactionServiceFactory()
    : ProfileKeyedServiceFactory("ReactionService") {}

ReactionServiceFactory::~ReactionServiceFactory() = default;

KeyedService* ReactionServiceFactory::BuildServiceInstanceFor(
    content::BrowserContext* context) const {
  return new ReactionService();
}

}  // namespace content_creation
